### ISSUE TITLE
feat(gauss_plume): Briggs plume rise for buoyant / momentum sources

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#82]
## Problem / Request
`plume_concentration` takes the effective release height `H` as an input. For flares, stacks and heated tanks the effective height is release height plus plume rise, which depends on the source's buoyancy/momentum flux, wind speed and stability. The roadmap lists `gauss_plume.plume_rise` (Briggs) as ☐; today users must precompute `H` outside the library and lose the gradient with respect to source parameters.

## User Story
> As a user inverting a flare emission, I want to specify stack height, exit velocity and temperature and have the effective height computed differentiably, so `H` isn't a hand-tuned constant.

## Proposed API
```python
# src/plumax/gauss_plume/plume_rise.py
def briggs_plume_rise(
    x: Float[Array, "N"], *, stack_height: float, exit_velocity: float, exit_temperature: float,
    stack_diameter: float, ambient_temperature: float, wind_speed: float, stability_class: str,
) -> Float[Array, "N"]: ...           # Δh(x); final rise past x_f
def effective_height(x, **kwargs) -> Float[Array, "N"]: ...   # stack_height + Δh(x)
```

## Implementation Steps
- [ ] Add `src/plumax/gauss_plume/plume_rise.py` with the Briggs (1969/1975) buoyant and momentum formulas, stability-dependent final-rise distance, pure JAX
- [ ] Accept an optional `effective_height` callable or array in `plume_concentration` / `simulate_plume` (keep scalar `H` path unchanged)
- [ ] Export from `gauss_plume/__init__.py`; add the API page entry

## Definition of Done
- [ ] Reproduces textbook Briggs rise values for canonical (F, u, class) inputs
- [ ] `jax.grad` w.r.t. `exit_temperature` is finite and matches finite differences
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/gauss_plume/test_plume_rise.py::test_textbook_values`
- [ ] `tests/gauss_plume/test_plume_rise.py::test_grad_matches_fd`

## Relationships
- Parent (theme epic): #70
